File: 백준/Bronze/9093. 단어 뒤집기/단어 뒤집기.cc
#include <iostream>
#include <string>
#include <stack>
using namespace std;

int main(void) {
	int T;
	cin >> T;
	cin.ignore();
	stack <char> mystack;
	string sentence;

	for (int i = 0; i < T; i++) {
		getline(cin, sentence);
		sentence += ' ';

		for (int i = 0; i < sentence.size(); i++) {
			if (sentence[i] == ' ') {
				while (!mystack.empty()) {
					cout << mystack.top();
					mystack.pop();
				}
				cout << ' ';
				
			}
			else mystack.push(sentence[i]);
		}
		cout << '\n';
	}

}

File: 백준/Bronze/9093. 단어 뒤집기/단어 뒤집기.py
T = int(input())
#T를 입력받음
stack = []
for _ in range(T):
    string = input()
    string +=' '
    reversed_string = ''
    for char in string:
        if char!=' ':
            stack.append(char)
            #공백이 아니면 글자추가함
        else:
            while stack:
                reversed_string += stack.pop()
            reversed_string += ' '
    print(reversed_string)